tls/extensions/s2n_client_psk.c

```diff
@@ -99,15 +109,13 @@ static S2N_RESULT s2n_match_psk_identity(struct s2n_array *known_psks, const str
         GUARD_RESULT(s2n_array_get(known_psks, i, (void**)&psk));
         ENSURE_REF(psk);
 
-        if (wire_identity->size != psk->identity.size) {
-            continue;
-        }
-
         ENSURE_REF(psk->identity.data);
         ENSURE_REF(wire_identity->data);
-        if (memcmp(psk->identity.data, wire_identity->data, wire_identity->size) == 0) {
+
+        uint32_t compare_size = MIN(wire_identity->size, psk->identity.size);
+        if (s2n_constant_time_equals(psk->identity.data, wire_identity->data, compare_size)
+            & (psk->identity.size == wire_identity->size) & (!*match)) {
```

Comment:
Q: Is it possible to add a unit test to confirm that the `s2n_match_psk_identity` function is indeed constant time and not leaking any timing information?

Reply:
Not with a unit test. That's what sidetrail is for. However, it doesn't look like we write sidetrail tests for every constant time function, and this definitely isn't a particularly important one. If we're going to write more sidetrail tests, there's probably a better target :)

Reply:
I was looking for a backlog task for side-trail and couldn't find one, it may be worth adding an issue to our backlog for side-trail updates/additions including this one :)